lib/models/sagemaker-schedule.ts

```diff
@@ -0,0 +1,48 @@
+import * as cdk from 'aws-cdk-lib';
+import * as scheduler from 'aws-cdk-lib/aws-scheduler';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as sagemaker from "aws-cdk-lib/aws-sagemaker";
+import { Construct } from 'constructs';
+
+export function createStartSchedule(scope: Construct, id: string, sagemakerEndpoint: sagemaker.CfnEndpoint, role: iam.Role, scheduleExpression: string) {
+  const startSchedule = new scheduler.CfnSchedule(scope, `${id}-${sagemakerEndpoint.endpointName}-startScheduler`, {
+    name: `StartScheduler-${id}-${sagemakerEndpoint.endpointName}`,
+    description: `created to start ${sagemakerEndpoint.endpointName}`,
+    flexibleTimeWindow: {
+      maximumWindowInMinutes: 5,
+      mode: 'FLEXIBLE',
+    },
+    scheduleExpression: `cron(${scheduleExpression})`,
+    state: 'ENABLED',
+    target: {
+      arn: 'arn:aws:scheduler:::aws-sdk:sagemaker:createEndpoint',
+      input: JSON.stringify({
+        EndpointName: sagemakerEndpoint.endpointName,
+        EndpointConfigName: sagemakerEndpoint.endpointConfigName,
+      }),
+      roleArn: role.roleArn,
+    },
+  });
+  return startSchedule;
+}
+
+export function createStopSchedule(scope: Construct, id: string, sagemakerEndpoint: sagemaker.CfnEndpoint, role: iam.Role, scheduleExpression: string) {
```

Comment:
Same comment on the props usage here as above - further to that though, could this just be a `createSchedule` call that takes a property like `scheduleType: 'START' | 'STOP'` that you can use to toggle on the differences? I only see the description and the target ARN that differ - could save you duplicating some of the rest of the logic.